amr-wind/wind_energy/actuator/turbine/fast/FastIface.cpp

```diff
@@ -282,9 +288,54 @@ void FastIface::fast_replay_turbine(FastTurbine& fi)
 #endif
 }
 
-void FastIface::fast_restart_turbine(FastTurbine&)
+void FastIface::fast_restart_turbine(FastTurbine& fi)
 {
     BL_PROFILE("amr-wind::FastIface::restart_turbine");
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        amrex::FileSystem::Exists(fi.checkpoint_file + ".chkp"),
+        "FastIface: Cannot find OpenFAST checkpoint file: "
+        + fi.checkpoint_file);
+
+    int abort_lev;
+    char chkpt_file[fast_strlen()];
+    copy_filename(fi.checkpoint_file, chkpt_file);
+
+    fast_func(
+        FAST_OpFM_Restart, &fi.tid_local, chkpt_file, &abort_lev,
+        &fi.dt_fast, &fi.num_blades, &fi.num_blade_elem, &fi.time_index,
+        &fi.to_cfd, &fi.from_cfd, &fi.to_sc, &fi.from_sc);
+
+    {
+#ifdef AMR_WIND_USE_OPENFAST
+        // Check if OpenFAST has tower and reset tower nodes appropriately
+        const int npts = fi.to_cfd.fx_Len;
+        const int nrotor_pts = fi.num_blades * fi.num_pts_blade + 1;
+        if (nrotor_pts == npts) {
+            amrex::OutStream()
+                << "OpenFAST model does not include tower for turbine: "
+                << fi.tlabel << " Turning off tower actuator points"
+                << std::endl;
+            fi.num_pts_tower = 0;
+        }
+        AMREX_ALWAYS_ASSERT(npts == (nrotor_pts + fi.num_pts_tower));
+#endif
+    }
+
+    // Determine the number of substeps for FAST per CFD timestep
+    fi.num_substeps = static_cast<int>(std::floor(fi.dt_cfd / fi.dt_fast));
+
+    AMREX_ALWAYS_ASSERT(fi.num_substeps > 0);
+    // Check that the time step sizes are consistent and FAST advances at an
+    // integral multiple of CFD timestep
+    double dt_err =
+        fi.dt_cfd / (static_cast<double>(fi.num_substeps) * fi.dt_fast) - 1.0;
+    if (dt_err > 1.0e-4) {
+        amrex::Abort(
+            "FastIFace: OpenFAST timestep is not an integral "
+            "multiple of CFD timestep");
+    }
+    
```

Comment:
seems duplicated from another method in this file. Would make sense to pull this out into a standalone function and use it in both places.